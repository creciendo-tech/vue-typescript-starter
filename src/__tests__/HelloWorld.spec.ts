import { mount } from "@vue/test-utils";
import Hello from "../componentes/HelloWorld.vue";

describe("HelloWorld", () => {
    it("renders properly", () => {
        const wrapper = mount(Hello, {
            props: { msg: "Hello Vitest" },
        });

        expect(wrapper.findComponent(Hello).isVisible()).toBe(true);
    });

    it("renders props", () => {
        const wrapper = mount(Hello, { props: { msg: "Hello Vitest" } });
        expect(wrapper.text()).toContain("Hello Vitest");
    });
    // test clicks on button changes count
    it("increments count when button is clicked", async () => {
        const wrapper = mount(Hello, { props: { msg: "Hello Vitest" } });
        expect(wrapper.text()).toContain("Hello Vitest");
        expect(wrapper.find("button").text()).toContain("count is 0");
        expect(wrapper.html()).toMatchSnapshot();
        await wrapper.find("button").trigger("click");
        expect(wrapper.find("button").text()).toContain("count is 1");
        expect(wrapper.html()).toMatchSnapshot();
        await wrapper.find("button").trigger("click");
        expect(wrapper.find("button").text()).toContain("count is 2");
        expect(wrapper.text()).toContain("Hello Vitest");
        expect(wrapper.html()).toMatchSnapshot();
        await wrapper.find("button").trigger("click");
        expect(wrapper.find("button").text()).toContain("count is 3");
        expect(wrapper.text()).toContain("Hello Vitest");
        expect(wrapper.html()).toMatchSnapshot();
        await wrapper.find("button").trigger("click");
    });
});
