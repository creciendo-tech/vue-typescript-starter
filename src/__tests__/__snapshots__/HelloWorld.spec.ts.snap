// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > increments count when button is clicked 1`] = `
"<h1 data-test="titulo">Hello Vitest</h1>
<div class="card"><button data-test="incremento" type="button">count is 0</button>
  <p> Edit <code>components/HelloWorld.vue</code> to test HMR </p>
</div>
<p> Check out <a href="https://vuejs.org/guide/quick-start.html#local" target="_blank">create-vue</a>, the official Vue + Vite starter </p>
<p> Install <a href="https://github.com/vuejs/language-tools" target="_blank">Volar</a> in your IDE for a better DX </p>
<p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>"
`;

exports[`HelloWorld > increments count when button is clicked 2`] = `
"<h1 data-test="titulo">Hello Vitest</h1>
<div class="card"><button data-test="incremento" type="button">count is 1</button>
  <p> Edit <code>components/HelloWorld.vue</code> to test HMR </p>
</div>
<p> Check out <a href="https://vuejs.org/guide/quick-start.html#local" target="_blank">create-vue</a>, the official Vue + Vite starter </p>
<p> Install <a href="https://github.com/vuejs/language-tools" target="_blank">Volar</a> in your IDE for a better DX </p>
<p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>"
`;

exports[`HelloWorld > increments count when button is clicked 3`] = `
"<h1 data-test="titulo">Hello Vitest</h1>
<div class="card"><button data-test="incremento" type="button">count is 2</button>
  <p> Edit <code>components/HelloWorld.vue</code> to test HMR </p>
</div>
<p> Check out <a href="https://vuejs.org/guide/quick-start.html#local" target="_blank">create-vue</a>, the official Vue + Vite starter </p>
<p> Install <a href="https://github.com/vuejs/language-tools" target="_blank">Volar</a> in your IDE for a better DX </p>
<p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>"
`;

exports[`HelloWorld > increments count when button is clicked 4`] = `
"<h1 data-test="titulo">Hello Vitest</h1>
<div class="card"><button data-test="incremento" type="button">count is 3</button>
  <p> Edit <code>components/HelloWorld.vue</code> to test HMR </p>
</div>
<p> Check out <a href="https://vuejs.org/guide/quick-start.html#local" target="_blank">create-vue</a>, the official Vue + Vite starter </p>
<p> Install <a href="https://github.com/vuejs/language-tools" target="_blank">Volar</a> in your IDE for a better DX </p>
<p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>"
`;
